lib/ansible/module_utils/apt_utils.py

```diff
@@ -0,0 +1,90 @@
+# -*- coding: utf-8 -*-
+
+# Copyright: (c) 2012, Flowroute LLC
+# Copyright: (c) 2018, Salvatore Mesoraca <s.mesoraca16@gmail.com>
+# Based on apt module written by Matthew Williams <matthew@flowroute.com>
+
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
```

Comment:
we really prefer to use BSD on module_utils for 3rd party module distribution, if any of my code is involved, this also counts as my agreement to relicense

Reply:
I have no problem doing this, if @mgwilliams agrees.

Reply:
While I was waiting for a response from Matthew Williams, I realized that it's not possible to re-license this module. It uses python-apt, which is GPL-2+ (http://changelogs.ubuntu.com/changelogs/pool/main/p/python-apt/python-apt_1.6.0/copyright) and so it can't be released with a BSD license.

Reply:
Legal just updated me so I'm going to update this comment thread (even though I don't know if this PR will be worked on in the future or not):

It is fine for this file to be licensed under the BSD if the authors agree to that.  The file itself is BSD in case someone ever extracts the code to use in a different context or in case ideas from the code were to be copied to another module_utils file.  When used together with the python-apt library, the work as a whole is used under the terms of the GPLv2+.

Legal does recommend that when cases like this arise that have the potential to confuse the users of the code, an explicit comment is desirable to warn of the confusion.  For instance, adding a comment like the following right after the BSD license header:

``` python
# Note: Although this code is BSD-licensed, it imports a module that is licensed under the GPL.
```